### DISCUSSION TITLE
Not showing FQDN URL when looking at Devices under Racks

### DISCUSSION THREAD

**Discussion by lewiryan** [netbox-community/netbox#5697] [Q&A] [answered]
### Environment
* Python version: 3.6.8
* NetBox version: 2..10.3

### Steps to Reproduce
1. Browse to one of the racks you have in Netbox -- /dcim/racks/1/
2. When attempting to click on a device in the rack it shows the IP address instead of FQDN of the server (x.x.x.x/dcim/devices/25/

### Expected Behavior

Expected the FQDN to show instead of the IP address

### Observed Behavior

Netbox shows the IP address instead of the FQDN, this can cause issues with TLS certificates if the IP address is not in the Subject Alternative Name in the TLS certificate

**Comment by jeremystretch** [comment]
You need to configure your HTTP frontend (e.g. nginx or Apache) to forward the requested hostname to NetBox. NetBox ships with example configuration files for both [nginx](https://github.com/netbox-community/netbox/blob/develop/contrib/nginx.conf) and [Apache](https://github.com/netbox-community/netbox/blob/develop/contrib/apache.conf).

**Reply by lewiryan** [reply]
So I do have that already installed with config files and the URL works as expected, its only this one section of Netbox it does not seem to apply.

When I go /dcim/racks/{RackID} and look at the elevation of a specific rack and click on a device in the rack I get the IP address of the netbox host instead of the FQDN

If I am going to add a device in that specific rack on the page /dcim/racks/{RackID} I get the FQDN of the host.

I'm able to browse all and select all devices under /dcim/devices/{DeviceID}
 
Attaching a picture to hopefully explain this problem.

Thanks
![netbox-bug](https://user-images.githubusercontent.com/12180089/105912327-89918980-5fe8-11eb-9e36-ab3502665a21.png)

**Comment by lewiryan** [comment, marked as answer]
Looking into this further,
My nginx config file had the following under sites-available/netbox

This config file had an IP address followed by the FQDN name, this was working before but perhaps an update to nginx caused this issue?

Original (modified example below)

`server_name 172.16.5.34 networks.example.com`

Changed:

`server_name networks.example.com`

I'm now able to able to select the device in the rack and it shows the FQDN.

Thanks

**Reply by candlerb** [reply]
If you'd used the nginx config that Jeremy linked to, it would have included:

```
        proxy_set_header X-Forwarded-Host $http_host;
```

and that would avoid the problem (Netbox generates URLs using whatever host name was in the incoming request, but it can't do this via a proxy unless it sets the X-Forwarded-Host header)
